common/lib/dependabot/update_checkers/base.rb

```diff
@@ -241,7 +241,7 @@ def updated_dependency_without_unlock
       sig { returns(Dependabot::Dependency) }
       def updated_dependency_with_own_req_unlock
         version = preferred_resolvable_version.to_s
-        previous_version = latest_resolvable_previous_version(version)&.to_s
+        previous_version = latest_resolvable_previous_version(version)
```

Comment:
I'm actually a little surprised, because:

```ruby
irb(main):001:0> nil&.to_s
=> nil
irb(main):002:0> nil.to_s
=> ""
```

Reply:
Thanks @jurre I'm also a bit stumped... This would suggest that `dependency.version` is returning `""`, but `Dependency` will raise if you try to create one with `previous_version = "", so I'm not sure how we could get into this state.

Reply:
My first thought was this is a base class so maybe a subclass is overriding `latest_resolvable_previous_version` and returning an empty string, but the only ecosystem that overrides it is npm and we're seeing this error for composer, nuget, and others.

Reply:
It looks like you can create a `Dependabot::Dependency` with a version of nil, then we call `to_s` on that?

https://github.com/dependabot/dependabot-core/blob/0ec3fe82cdc08567755ba023267c1b4fe062dde1/common/lib/dependabot/version.rb#L17

So we end up with it being `""` through that path?

Reply:
Yep, I think that's it!

Reply:
Ecosystems implement their own version classes and it seems most call `#to_s` on the version argument, plus `Gem::Version` is going to call `#to_s` anyways, so we'd need to override `Version#to_s` and `Version#version` and methods on ecosystem version classes that return the stringified version.

Instead of going down that rabbit hole, I think the most sensible thing to do is, instead of raising, just have `Dependency` set its `version` or `previous_version` to nil if either is an empty string.